src/rpc/common/Concepts.h

```diff
@@ -49,12 +49,17 @@ concept Requirement = requires(T a) {
  */
 // clang-format off
 template <typename T>
-concept Handler = requires(T a, typename T::Input in, typename T::Output out) {
+concept Handler = 
+((requires(T a, typename T::Input in, typename T::Output out) {
     { a.spec() } -> std::same_as<RpcSpecConstRef>;
```

Comment:
i think we need the spec in both cases, right?
also, i'd split into separate smaller concepts and then combine into one big concept using smaller ones for readability :)

Reply:
Actually, nvm. you are right, we need it only for when we actually have some input to validate.
Please try to split into smaller concepts and built on composition - overall, looks like a good solution to me 👍